### PR TITLE
Propose initial API for discussion

### PR DESCRIPTION
**Pull request by rohantmp**
To view API changes, you only need to look at the first commit. The others are just generated files

Yaml example: https://github.com/rohantmp/local-disk-operator/blob/apiPrototype/config/samples/lvm_v1alpha1_topolvmcluster.yaml

Go definition: https://github.com/rohantmp/local-disk-operator/blob/apiPrototype/api/v1alpha1/topolvmcluster_types.go#L29

### CODE REVIEW

File: api/v1alpha1/topolvmcluster_types.go [hunk 1]
```diff
@@ -28,8 +30,62 @@ type TopoLVMClusterSpec struct {
...
+}
+
+// DeviceSelector matches if all the rules match.
+// (ruleType1 AND ruleType2 AND ... AND ruleTypeN-1 AND ruleTypeN)
+// Each characteristic is represented either by a ruleType, a ruleType is either a list of OR'd rules or a single rule
+type DeviceSelector struct {
+	// MaxMatchesPerNode to allow a use-cases where not all matches on a node are consumed.
+	// +optional
+	MaxMatchesPerNode *int `json:"maxMatchesPerNode,omitempty"`
+	// Devices is the list of devices that should be used for automatic detection.
```
Comment (jmolmo) [inline api/v1alpha1/topolvmcluster_types.go:58]: Typo: Device types

File: api/v1alpha1/topolvmcluster_types.go [hunk 2]
```diff
@@ -28,8 +30,62 @@ type TopoLVMClusterSpec struct {
...
+	// This would be one of the types supported by the local-storage operator. Currently,
+	// the supported types are: disk, part. If the list is empty only `disk` types will be selected
+	// +optional
+	DeviceTypes []DeviceType `json:"deviceTypes,omitempty"`
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
```
Comment (jmolmo) [inline api/v1alpha1/topolvmcluster_types.go:63]: If the attribute is just a boolean, why to use a slice of strings...?

File: api/v1alpha1/topolvmcluster_types.go [hunk 3]
```diff
@@ -28,8 +30,62 @@ type TopoLVMClusterSpec struct {
...
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperties []DeviceMechanicalProperty `json:"deviceMechanicalProperties,omitempty"`
```
Comment (jmolmo) [inline api/v1alpha1/topolvmcluster_types.go:66]: `DeviceMechanicalProperties`
If the attribute is just a boolean, why to use a slice of strings...? (If not provided we can select both types of disk.)

File: config/crd/bases/lvm.openshift.io_topolvmclusters.yaml [hunk 4]
```diff
@@ -0,0 +1,217 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+  creationTimestamp: null
+  name: topolvmclusters.lvm.openshift.io
```
Comment (jmolmo) [inline config/crd/bases/lvm.openshift.io_topolvmclusters.yaml:9]: I think we should avoid explicit references to openshift (or any other RH product) . IMHO one of main targets should be to produce an operator which will be easily included under the topolvm umbrella.

Comment (rohantmp) [reply config/crd/bases/lvm.openshift.io_topolvmclusters.yaml:9]: Sounds good! Looks like topolvm-operator [might be planning to switch to topolvm.io](https://github.com/topolvm/topolvm/issues/168)

File: api/v1alpha1/topolvmcluster_types.go (hunk 3 above)
Comment (rohantmp) [reply api/v1alpha1/topolvmcluster_types.go:66]: We could select only-hdd or only-sdd or both with this setting

**Pull request closed without merging**